From: Auto-Generated
Date: Mon, 17 Feb 2026 00:00:00 +0000
Subject: [PATCH] Fix mctpwplus API changes in main.cpp

- Update endpoint map iteration from getEndpointMap() to getEndpointMapExtended()
- Update Event struct usage from evt.eid to evt.deviceId.mctpEID()

---
 main.cpp | 11 ++++++-----
 1 file changed, 6 insertions(+), 5 deletions(-)

diff --git a/main.cpp b/main.cpp
index 1234567..bcdefgh 100644
--- a/main.cpp
+++ b/main.cpp
@@ -70,7 +70,8 @@ struct App
                 mctpWrappers.emplace(bindingType, wrapper);
                 wrapper->detectMctpEndpoints(yield);
-                for (auto& [eid, service] : wrapper->getEndpointMap())
+                for (auto& deviceId : wrapper->getEndpointMapExtended())
                 {
+                    auto eid = deviceId.mctpEID();
                     try
                     {
                         auto drive = std::make_shared<nvmemi::Drive>(
@@ -286,7 +287,7 @@ bool DeviceUpdateHandler::createDrive(mctpw::eid_t eid)
                     boost::system::error_code ec;
                     timer.async_wait(yield[ec]);
                     retryCount--;
-                    driveCreated = createDrive(evt.eid);
+                    driveCreated = createDrive(evt.deviceId.mctpEID());
                     if (!driveCreated)
                     {
                         timer.expires_from_now(boost::posix_time::millisec(300));
@@ -302,13 +303,13 @@ bool DeviceUpdateHandler::createDrive(mctpw::eid_t eid)
         break;
         case mctpw::Event::EventType::deviceRemoved: {
-            if (app.drives.erase(evt.eid) == 1)
+            if (app.drives.erase(evt.deviceId.mctpEID()) == 1)
             {
                 phosphor::logging::log<phosphor::logging::level::INFO>(
                     "Drive removed",
-                    phosphor::logging::entry("EID=%d", evt.eid));
+                    phosphor::logging::entry("EID=%d", static_cast<int>(evt.deviceId.mctpEID())));
             }
             else
             {
                 phosphor::logging::log<phosphor::logging::level::ERR>(
                     "No drive found mapped to eid",
-                    phosphor::logging::entry("EID=%d", evt.eid));
+                    phosphor::logging::entry("EID=%d", static_cast<int>(evt.deviceId.mctpEID())));
             }
